tests/python/relay/test_op_level2.py

```diff
@@ -517,6 +517,45 @@ def test_upsampling():
     _test_upsampling("NHWC", "BILINEAR")
 
 
+def test_conv2d_int8_intrinsics():
+    def _compile(input_dtype, weight_dtype, output_dtype):
+        n, ic, h, w, oc, ch, cw = 1, 16, 224, 224, 32, 3, 3
+        x = relay.var("x", relay.TensorType((n, ic, h, w), input_dtype))
+        w = relay.var("w", relay.TensorType((oc, ic, ch, cw), weight_dtype))
+        y = relay.nn.conv2d(x, w,
+                            kernel_size=(ch, cw),
+                            channels=oc,
+                            padding=(1, 1),
+                            dilation=(1, 1),
+                            out_dtype=output_dtype)
+        func = relay.Function([x, w], y)
+        wdata = np.random.rand(oc, ic, ch, cw) * 10
+        parameters = {"w": tvm.nd.array(wdata.astype(weight_dtype))}
+        with relay.build_config(opt_level=3):
+            graph, lib, params = relay.build(func, target, params=parameters)
+        assembly = lib.get_source("asm")
+        return assembly
+
+    # compile conv2d for x86 (skylake) and test assembly contains *pmadd* instructions
+    target = "llvm -mcpu=skylake-avx512"
+    name = "llvm.x86.avx512.pmaddubs.w.512"
+    llvm_id = tvm.codegen.llvm_lookup_intrinsic_id(name)
+    if llvm_id != 0:
+        # Intel Int8 instruction need uint8 data and int8 kernel
+        asm = _compile(input_dtype="uint8",
+                       weight_dtype="int8",
+                       output_dtype="int32")
+        # Check that intrinisic is present in the assembly.
+        assert "pmadd" in asm
+
+        # Ensure that code is generated when datatypes are not HW supported.
+        asm = _compile(input_dtype="int8",
+                       weight_dtype="int8",
+                       output_dtype="int32")
+        # Check that intrinisic is not present in the assembly.
+        assert "pmadd" not in asm
+
```

Comment:
Because we have make int8 schedule using existing FP32 schedule, we have vectorize support. I think we should supply another one assert, i.e. vectorize multiply / add int (we could add another one target, for example `llvm -mcpu=core-avx2`,  to make sure we have vectorize support).  I can not make sure x86's correct instruction. Maybe `vpmulld ` ?

Reply:
Added